app.js

```diff
@@ -143,37 +123,28 @@ function _map(array, callbackFunction) {
 	return arrMaped;
 }
 
-function _indexOf(array, value) {
-	for (let i = 0; i < array.length; i++) {
-		if (array[i] === value) {
+Array.prototype._indexOf = function(value) {
+	for (let i = 0; i < this.length; i++) {
+		if (this[i] === value) {
 			return i;
 		}
 	}
 	return -1;
-}
+};
 
-function _indexOfRecursive(array, value) {
-	if (!array.length) return -1;
-	return array.pop() === value ? array.length : _indexOfRecursive(array, value);
-}
+Array.prototype._indexOfRecursive = function(value) {
+	if (!this.length) return -1;
+	return this.pop() === value ? this.length : this._indexOfRecursive(value);
+};
 
-function _includes(array, value) {
-	return _indexOf(array, value) !== -1;
-}
+Array.prototype._includes = function(value) {
+	return this._indexOf(value) !== -1;
+};
 
-function _intersection(array1, array2) {
-	const repetedIntersection = fakeFilter(array1, x => fakeIncludes(array2, x));
-	return removeDuplicatesFrom(repetedIntersection);
-}
+Array.prototype._intersection = function(array1, array2) {
+	return Array.from(new Set(array1._filter(x => array2._includes(x))));
+};
 
-function _union(arr1, ...arr2) {
-	const arrUnion = arr1.slice();
-	_forEach(arr2, function(i) {
-		_forEach(i, function(j) {
-			if (!_includes(arrUnion, j)) {
-				arrUnion.push(j);
-			}
-		});
-	});
-	return arrUnion;
-}
+Array.prototype._union = function(array1, array2) {
```

Comment:
Array.prototype._union = function(array2) {
	return Array.from(new Set([...this, ...array2]));
};